src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedDeletedMethod.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a method that has been deleted in an Edit and Continue session, and whose body
+    /// throws a System.MissingMethodException.
+    /// </summary>
+    internal sealed class SynthesizedDeletedMethod : MethodSymbol, ISynthesizedDeletedMethod
+    {
+        private readonly MethodSymbol _methodSymbol;
+        private readonly NamedTypeSymbol _containingTypeSymbol;
+
+        internal SynthesizedDeletedMethod(MethodSymbol methodSymbol, NamedTypeSymbol containingTypeSymbol)
+            : base()
+        {
+            _methodSymbol = methodSymbol;
+            _containingTypeSymbol = containingTypeSymbol;
+        }
+
+        // MethodSymbol properties that delegate through _containingTypeSymbol
+
+        internal override ModuleSymbol ContainingModule => _containingTypeSymbol.ContainingModule;
+
+        public override Symbol ContainingSymbol => _containingTypeSymbol;
```

Comment:
If the deleted method is a local function, will `ContainingSymbol` be correct here?

Reply:
I have made no attempt at making this work in all circumstances yet 😛